src/test-helper.js

```diff
@@ -667,64 +639,68 @@ function verifyAccessDenied(doc, oldDoc, expectedAuthorization) {
   requireRole.throwWith(roleAccessDeniedError);
   requireUser.throwWith(userAccessDeniedError);
 
+  var syncFuncError = null;
   try {
     syncFunction(doc, oldDoc);
-    assert.fail('Document authorization succeeded when it was expected to fail');
```

Comment:
If this statement executes, it throws an exception that is subsequently caught by the `catch` block and mishandled. The result is now verified outside of the try-catch to avoid the issue.